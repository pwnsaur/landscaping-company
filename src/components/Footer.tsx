import styled from 'styled-components';

const Footer = () => {
  return (
    <FooterContainer>
      <FooterText>Copyright 2023 Hackerman</FooterText>
    </FooterContainer>
  );
};

export default Footer;

const FooterContainer = styled.footer`
  display: flex;
  width: 100%;
  height: 15rem;
  justify-content: center;
  align-items: center;
  margin: auto 0 0;
  background-color: ${({ theme }) => theme.colors.darkGreen};
  /* z-index: 1; */
`;

const FooterText = styled.p`
  color: white;
`;
