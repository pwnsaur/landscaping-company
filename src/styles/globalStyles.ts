import { createGlobalStyle } from 'styled-components';
import { Nunito } from 'next/font/google';

const nunito = Nunito({
  subsets: ['latin', 'latin-ext'],
  display: 'swap',
  fallback: ['sans-serif'],
});

const GlobalStyles = createGlobalStyle`
* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: ${nunito.style.fontFamily}, sans-serif;
}

.content {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  width: 100%;
  align-items: center;
  justify-content: center;
  background-color: #ededed;
}

a {
  color: inherit;
  text-decoration: none;
}
`;

export default GlobalStyles;
