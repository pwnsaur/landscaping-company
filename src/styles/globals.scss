@import url('https://fonts.googleapis.com/css2?family=Nunito&display=swap');

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: Nunito, sans-serif;
}

// .container {
//   width: 100%;
// }

a {
  color: inherit;
  text-decoration: none;
}
