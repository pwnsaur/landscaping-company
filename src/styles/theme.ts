export const theme = {
  background: '#ededed',
  secondary: 'green',
  font: 'Arial',
};
