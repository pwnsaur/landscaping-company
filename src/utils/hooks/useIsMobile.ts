import useWindowSize from './useWindowSize';

const useIsMobile = (): boolean => {
  const { width, height } = useWindowSize();
  const currentWidth = width || 0;
  const currentHeight = height || 0;
  const isMobile = currentWidth <= 640;

  return isMobile;
};

export default useIsMobile;
