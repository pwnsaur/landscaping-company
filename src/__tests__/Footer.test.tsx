import { render, screen } from '@testing-library/react';
import Footer from '@/components/Footer';
import { ThemeProvider } from 'styled-components';
import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';

function TestWrapper() {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>
      <Footer />
    </ThemeProvider>
  );
}

describe('Footer snapshot', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(<TestWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Footer copyright', () => {
  test('say my name', () => {
    render(<TestWrapper />);
    const copyrightText = screen.getByText(/Copyright 2023 Hackerman/i);
    expect(copyrightText).toBeInTheDocument();
  });
});
