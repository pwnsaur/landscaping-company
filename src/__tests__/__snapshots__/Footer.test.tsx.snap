// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="sc-f0956a4b-0 joZgVz"
  >
    <p
      class="sc-f0956a4b-1 gaAhTs"
    >
      Copyright 2023 Hackerman
    </p>
  </footer>
</DocumentFragment>
`;
